src/lcm/dispatchers.py

```diff
@@ -40,42 +50,53 @@ def spacemap(func, dense_vars, sparse_vars, *, dense_first):
             above but there might be additional dimensions.
 
     """
-    func = allow_args(func)  # vmap cannot deal with keyword-only arguments
-
+    # Check inputs
+    # ==================================================================================
     if not set(dense_vars).isdisjoint(sparse_vars):
         raise ValueError("dense_vars and sparse_vars overlap.")
 
-    _all_variables = dense_vars + sparse_vars
-    if len(_all_variables) != len(set(_all_variables)):
-        raise ValueError("Same argument provided more than once.")
+    if len(set(dense_vars)) < len(dense_vars):
```

Comment:
```suggestion
    if duplicates := {v for v in dense_vars if dense_vars.count(v) > 1}:
```
*(untested)* Same below.